File: 67.add-binary.py
class Solution:
    def addBinary(self, a, b):
        """
        :type a: str
        :type b: str
        :rtype: str
        """
        if len(a) < len(b):
            return self.addBinary(b, a)
        c = 0
        sum = ""
        j = len(b) - 1
        for i in range(len(a) - 1, -1, -1):
            if a[i] == '1':
                c += 1
            if j >= 0 and b[j] == '1':
                c += 1
            if c % 2 == 1:
                sum = '1' + sum
            else:
                sum = '0' + sum
            c = c // 2
            j -= 1
        if c:
            return "1" + sum
        else:
            return sum;

File: 67.add-binary.js
/**
 * @param {string} a
 * @param {string} b
 * @return {string}
 */
var addBinary = function(a, b) {
    if (a.length < b.length) {
        return addBinary(b, a);
    }
    let c = 0;
    let sum = "";
    for (let i = a.length - 1, j = b.length - 1; i >= 0; i--, j--) {
        if (a[i] == '1') {
            c++;
        }
        if (j >= 0 && b[j] == '1') {
            c++;
        }
        sum = (c % 2 ? '1' : '0') + sum;
        c = c >> 1;
    }
    if (c) {
        return "1" + sum;
    } else {
        return sum;
    }
};